src/arith/rewrite_simplify.cc

```diff
@@ -1667,6 +1667,22 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const CallNode* op) {
       return match.value();
     }
   }
+  if (op->op.same_as(tir::builtin::if_then_else())) {
+    // Simplify nested if_then_else
+    const PrimExpr& cond = op->args[0];
+    const PrimExpr& then_expr = op->args[1];
+    const PrimExpr& else_expr = op->args[2];
+    const CallNode* inner_call = then_expr.as<CallNode>();
+    if (inner_call != nullptr && inner_call->op.same_as(tir::builtin::if_then_else())) {
```

Comment:
Overall LGTM. I wonder if this implies that only one level of nested if_then_else could exist in current TIR structure.

Reply:
it supports multiple levels, as this mutator is recursive (`then_expr` here is from the recursion result)